test/TestProjects/FirstTest-Cadl/Generated/Models/Thing.cs

```diff
@@ -15,15 +15,25 @@ public partial class Thing
     {
         /// <summary> Initializes a new instance of Thing. </summary>
         /// <param name="name"></param>
-        /// <exception cref="ArgumentNullException"> <paramref name="name"/> is null. </exception>
-        public Thing(string name)
+        /// <param name="requiredUnion"></param>
+        /// <param name="requiredLiteral"></param>
+        /// <exception cref="ArgumentNullException"> <paramref name="name"/>, <paramref name="requiredUnion"/> or <paramref name="requiredLiteral"/> is null. </exception>
+        public Thing(string name, string requiredUnion, string requiredLiteral)
         {
             Argument.AssertNotNull(name, nameof(name));
+            Argument.AssertNotNull(requiredUnion, nameof(requiredUnion));
+            Argument.AssertNotNull(requiredLiteral, nameof(requiredLiteral));
 
             Name = name;
+            RequiredUnion = requiredUnion;
+            RequiredLiteral = requiredLiteral;
         }
 
         /// <summary> Gets or sets the name. </summary>
         public string Name { get; set; }
+        /// <summary> Gets or sets the required union. </summary>
+        public string RequiredUnion { get; set; }
```

Comment:
I remember we've decided not to support union type property.

Reply:
Yes, this is just to verify that emitter correct emit the type, after codegen support union, we will write another test, and remove here.